File: backend/src/main/java/com/logap/backend/Models/Produto.java
package com.logap.backend.Models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String localizacao;
    private float valor;
    private int quantidade;

    @ManyToOne
    @JoinColumn(name = "id_categoria")
    private Categoria categoria;

    @ManyToOne
    @JoinColumn(name = "id_fornecedor")
    private Fornecedor fornecedor;

}

File: frontend/src/models/Produto.ts
import { Categoria } from "./Categoria";
import { Fornecedor } from "./Fornecedor";

export class Produto {
    id: number;
    name: string;
    localizacao: string;
    valor: number;
    quantidade: number;
    categoria: Categoria;
    fornecedor: Fornecedor;

}
